src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LoweredDynamicOperationFactory.cs

```diff
@@ -698,7 +698,7 @@ private static NamedTypeSymbol CreateCallSiteContainer(SyntheticBoundNodeFactory
 
             if (factory.TopLevelMethod.IsGenericMethod)
             {
-                return synthesizedContainer.Construct(factory.TopLevelMethod.TypeParameters.Cast<TypeParameterSymbol, TypeSymbol>());
+                return synthesizedContainer.Construct(factory.TopLevelMethod.TypeParameters);
```

Comment:
>return synthesizedContainer.Construct(factory.TopLevelMethod.TypeParameters); [](start = 16, length = 77)

It is not clear what motivated this change and how it is related to the purpose of this PR

Reply:
This must have been committed by accident, will revert.